src/lms-harmonizer/edfi_lms_harmonizer/migrator.py

```diff
@@ -0,0 +1,134 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from os import path
+from typing import List
+
+from sqlalchemy.exc import ProgrammingError
+from sqlparse import split
+
+from edfi_sql_adapter.sql_adapter import Adapter, Statement
+
+
+logger = logging.getLogger(__name__)
+
+MIGRATION_SCRIPTS = [
+    # CAUTION: these scripts will run in order from "top to bottom", so it is
+    # critical to maintain the script order at all times.
+    "0001_initialize_lms_database",
+    "0002_harmonize_lmsuser_canvas",
+    "0003_harmonize_lmsuser_schoology",
+    "0004_harmonize_lmsuser_google-classroom",
+    "0005_harmonize_lmssection_canvas",
+    "0006_harmonize_lmssection_google_classroom",
+    "0007_harmonize_lmssection_schoology",
+    "0008_view_exceptions_LMSSection",
+    "0009_view_exceptions_LMSUser",
+    "0010_harmonize_assignment",
+    "0011_harmonize_assignment_submissions",
+]
+
+
+def _get_script_path(adapter: Adapter, script_name: str) -> str:
+    script_dir = path.join(path.dirname(__file__), "scripts", adapter.engine.name)
+    return path.join(script_dir, script_name)
+
+
+def _read_statements_from_file(full_path: str) -> List[str]:
+    raw_sql: str
+    with open(full_path) as f:
+        raw_sql = f.read()
+
+    statements: List[str] = split(raw_sql)
+    return statements
+
+
+def _script_has_been_run(adapter: Adapter, migration: str) -> bool:
+    try:
+        statement = f"SELECT 1 FROM lms.MigrationJournal_Harmonizer WHERE script = '{migration}';"
+        response = adapter.get_int(statement)
+
+        return bool(response == 1)
+    except ProgrammingError as error:
+        if (
+            # PostgreSLQ error
+            "psycopg2.errors.UndefinedTable" in error.args[0]
+            or
+            # SQL Server error
+            "Invalid object name" in error.args[0]
+        ):
+            # This means it is a fresh database where the migrationjournal table
+            # has not been installed yet.
+            return False
+
+        raise
+
+
+def _record_migration_in_journal(adapter: Adapter, migration: str) -> None:
+    statement = Statement(
+        f"INSERT INTO lms.MigrationJournal_Harmonizer (script) values ('{migration}');",
+        "Updating migration journal table"
+    )
+
+    adapter.execute([statement])
+
+
+def _lms_migration_journal_exists(adapter: Adapter) -> bool:
+    statement = """
```

Comment:
A difference from the DS Loader migrator, check for existence of the migration journal instead of just the lms schema, as lms schema is probably already there in a normal workflow.